packages/caravan-clients/src/client.test.ts

```diff
@@ -1823,4 +1825,327 @@ describe("BlockchainClient", () => {
+      const result = await client.getAddressTransactionHistory(testAddress, 20, 40);
+
+      expect(mockGet).toHaveBeenCalledWith(`/address/${testAddress}/txs?count=20&skip=40`);
+      expect(result).toHaveLength(1);
+    });
+
+    it('should use correct endpoint for BLOCKSTREAM', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.BLOCKSTREAM,
+        network: Network.MAINNET,
+      });
+
+      const mockResponse = [
+        { 
+          txid: 'tx1', 
+          fee: 1000,
+          timestamp: 1234567890,
+          version: 1,
+          locktime: 0,
+          vin: [],
+          vout: [],
+          size: 250,
+          weight: 1000,
+          status: { confirmed: true }
+        }
+      ];
+      const mockGet = vi.fn().mockResolvedValue(mockResponse);
+      client.Get = mockGet;
+
+      const result = await client.getAddressTransactionHistory(testAddress, 15, 30);
+
+      expect(mockGet).toHaveBeenCalledWith(`/address/${testAddress}/txs?limit=15&offset=30`);
+      expect(result).toHaveLength(1);
+    });
+  });
+
+  describe('error handling', () => {
+    it('should handle missing provider', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        network: Network.MAINNET,
+      });
+
+      (client as any).provider = undefined;
+
+      await expect(client.getAddressTransactionHistory(testAddress))
+        .rejects.toThrow('Provider must be specified for public clients');
+    });
+
+    it('should handle API errors', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
+
+      const mockGet = vi.fn().mockRejectedValue(new Error('Network error'));
+      client.Get = mockGet;
+
+      await expect(client.getAddressTransactionHistory(testAddress))
+        .rejects.toThrow('Failed to get address transaction history: Network error');
+    });
+
+    it('should handle invalid response format', async () => {
+      const client = new BlockchainClient({
+        type: ClientType.PUBLIC,
+        provider: PublicBitcoinProvider.MEMPOOL,
+        network: Network.MAINNET,
+      });
+
+      const mockGet = vi.fn().mockResolvedValue('invalid response');
+      client.Get = mockGet;
+
+      const result = await client.getAddressTransactionHistory(testAddress);
+      
+      // Should handle gracefully and return empty array
+      expect(result).toHaveLength(0);
+    });
+  });
+});
+
+describe('getWalletTransactionHistory', () => {
+  let mockCallBitcoindWallet: any;
+
+  beforeEach(() => {
+    mockCallBitcoindWallet = vi.spyOn(wallet, 'callBitcoindWallet');
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  describe('parameter validation', () => {
+    it('should validate count parameter', async () => {
+      const client = new BlockchainClient({
```

Comment:
pull this out. If you need to have flexibility with args, you can create a function `const getNewClient = (argOverrides) => new BlockChainClient({ ...defaultArgs, ...argOverrides })`